README.md

```diff
@@ -1,33 +1,174 @@
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
 ## Usage
+
 ```
-npm install @smg-automotive/example
+npm install @smg-automotive/api-client-pkg
 ```
 
+````typescript
+import { ApiClient } from "@smg-automotive/api-client-pkg"
+````
+
+### Configuration
+
+The available instance methods are listed below. The specified config will be merged with the instance config.
+
+````typescript
+ApiClient.configure({
+  baseUrl: 'https://api.automotive.ch/api',
+  headers: { 'Content-Type': 'application/json' },
+})
+````
+
+### Instance methods
+
+The ApiClient provides promise-based methods for `GET`, `POST`, `PUT` and `DELETE`.
+
+### Typing
+
+You can define the request and the response type of your method as follows:
+
+````typescript
+interface Car {
+  make: string
+  horsepower: number
+}
+
+// data has type Car
+var data = await ApiClient.get<Car>('/listings/search')
+
+// data has type { id: string } and body is typed as Car
+var data = await ApiClient.post<{ id: string }, Car>('/listings/create',
+  { make: 'bmw', horsepower: 300 },)
+````
+
+### Parameters
+
+If your request URL contains parameters, you can use curly brackets and the ApiClient replaces it with the passed data.
+This is aligned with the OpenAPI specification used by Swagger. Hence, you can easily search for occurrences of your
+APIs.
+
+````typescript
+// fetch is called with https://baseUrl.api.ch/dealers/123/listings/456
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
+})
+````
+
+### Authorization
+
+You can pass `accessToken` to the request options if you are talking to a protected API. The client will automatically
+set the header with a Bearer token.
+
+````typescript
+// fetch is called with the header { Authorization: `Bearer ${accessToken}` }
+ApiClient.post<{ id: number }, { name: string; listingIds: number[] }>(
+  "users/me/listing-comparisons",
+  { name: "test1", listingIds: [890163] },
+  { accessToken: authHeader.accessToken }
+)
+  .then((res) => {
+    // do something with the response
+  })
+  .catch((error) => {
+    // do something in case of an error
+  })
+````
+
+### Error handling
+
+The promise is rejected if any error occurs on API level. You can use promise chaining or `try/catch` to add error
+handling.
+
+````typescript
+ApiClient.post<LeasingCalculation, LeasingData>(
+  "listings/calculate-leasing",
+  {
+    downPayment: shouldWork ? 7300 : 17500,
+    duration: 48,
+    estimatedKmPerYear: shouldWork ? 10000 : 30000,
+    firstRegistrationDate: "2020-07-01",
+    residualValue: 12045,
+    price: 36500,
+  }
+)
+  .then((res) => {
+    // res is typed to LeasingCalculation
+  })
+  .catch((error: ResponseError) => {
+    // do any error handling you want
+  })
```

Comment:
```suggestion
```

The example below covers this and uses `async`/`await`

Reply:
yes the two examples do the exact same thing. I think promise chaining or async/await is preference. I wanted to add both to have it complete